### ISSUE TITLE
Using REBASE with a different PRM

### ISSUE THREAD

**Issue by mukhal** [thu-wyz/inference_scaling#3]
Thanks for the code! 

I'm trying to use your code with this [prm](https://huggingface.co/peiyi9979/math-shepherd-mistral-7b-prm), which I believe uses the same end-of-step token as the Llemma reward model you use. However, the scores returned from [this](https://github.com/thu-wyz/inference_scaling/blob/e411c0ae142ed57889c4e65a6a8fece1be708dc9/rebase.py#L107) line
 are empty and it's very hard to debug the sglang code. What could the issue be here? Any help is appreciated!

**Comment by mukhal** [comment]
i think i figured out the issue. Seems that the generator model is expected to generate the EOS token after each step. Since my generator model did not, this cause [this](https://github.com/thu-wyz/sglang/blob/b638847878f83d0eee3939653fb4c04c72acdc18/python/sglang/srt/layers/logits_processor.py#L36) to fail.

**Comment by zzjchen** [comment]
> i think i figured out the issue. Seems that the generator model is expected to generate the EOS token after each step. Since my generator model did not, this cause [this](https://github.com/thu-wyz/sglang/blob/b638847878f83d0eee3939653fb4c04c72acdc18/python/sglang/srt/layers/logits_processor.py#L36) to fail.

@mukhal Hi, could you explain how this can be solved? I ran into the same issue and don't know how to fix it.

**Comment by zzjchen** [comment]
Not exactly the same, I'm trying to call the API to the reward model, but always gets empty in the returned Response
``` python
>>> response=post_http_request(question,api_url,max_tokens=0,forward_only=True, logits_require_id=8094)
>>> response
<Response [200]>
>>> response.json()
{'text': '', 'meta_info': {'prompt_tokens': 134, 'completion_tokens': 1, 'completion_tokens_wo_jump_forward': 1, 'id': '991d3c70a25144b7a4e79c82470d6b0e'}, 'scores': [], 'forward_only': True}
```
Some variables:
``` python
question="How many positive whole-number divisors does 196 have?Step 1: The prime factorization of 196 is $2^2 \\cdot 7^2$. \u043a\u0438\nStep 2: To find the number of divisors, we consider the exponents of the prime factors. \u043a\u0438\nStep 3: For each prime factor, we have 3 choices: include 0, include 1, or include 2. \u043a\u0438\nStep 4: Therefore, the number of positive whole-number divisors is $(2+1)(2+1)=\\boxed{9}$.The answer is: 9 \u043a\u0438"
api_url="http://127.0.0.1:30010/generate" #30010 is where the reward model service is
```
